src/common/hooks/useNeedsZeroApproval.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react'
+
+import { Erc20 } from '@cowprotocol/abis'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+
+import { Nullish } from 'types'
+
+import { shouldZeroApprove as shouldZeroApproveFn } from './useShouldZeroApprove/shouldZeroApprove'
+
+export function useNeedsZeroApproval(
+  erc20Contract: Nullish<Erc20>,
+  spender: Nullish<string>,
+  sellAmount: Nullish<CurrencyAmount<Token>>
+): boolean {
+  const [shouldZeroApprove, setShouldZeroApprove] = useState(false)
+
+  useEffect(() => {
+    if (!erc20Contract || !spender || !sellAmount) return
+
+    shouldZeroApproveFn({
+      tokenContract: erc20Contract,
+      spender: spender,
+      amountToApprove: sellAmount,
+      isBundle: true,
+    }).then(setShouldZeroApprove)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [erc20Contract?.address, sellAmount?.quotient?.toString(), spender])
```

Comment:
why did you go for this `.quotient?.toString()` + eslint disabling?

Reply:
For stable reference
Safer to use undefined|string than and object.
Same for the contract address instead of the instance itself